src/pybind/mgr/volumes/fs/purge_queue.py

```diff
@@ -106,8 +106,21 @@ def __init__(self, volume_client, tp_size):
         self.vc = volume_client
         super(ThreadPoolPurgeQueueMixin, self).__init__(volume_client, "purgejob", tp_size)
 
+    def set_config_opt_disable_purge_trash(self, disable_purge_trash):
+        """
+        Set the value of the config option "disable_purge_trash" to the latest
+        value passed by the user. This allows not only stopping the purges in
+        the future but also allows halting ongoing purges midway.
+        """
+        self.fs_client.disable_purge_trash = disable_purge_trash
+        if self.fs_client.disable_purge_trash:
+            self.vc.purge_queue.cancel_all_jobs()
+
     def get_next_job(self, volname, running_jobs):
         return get_trash_entry_for_volume(self.fs_client, self.vc.volspec, volname, running_jobs)
 
     def execute_job(self, volname, job, should_cancel):
+        if self.fs_client.disable_purge_trash:
```

Comment:
While this will technically work, I don't think this is the approach this functionality should take. The threads would still be executing over and over again doing nothing, while we would want the threads to be blocked.

Reply:
>  threads to be blocked.

IIUC, you want the thread creation to be blocked (after `cancel_all_jobs()` has been run). Is that correct?

Reply:
yes